MoreLinq/Experimental/Await.cs

```diff
@@ -531,149 +610,82 @@ IEnumerable<TResult> _(int maxConcurrency, TaskScheduler scheduler, bool ordered
             }
         }
 
-        enum Notice { Result, Error, End }
-
-        static async Task CollectToAsync<T, TResult, TNotice>(
-            this IEnumerator<T> e,
-            Func<T, Task<TResult>> taskSelector,
-            BlockingCollection<TNotice> collection,
-            Func<T, Task<TResult>, TNotice> completionNoticeSelector,
-            Func<Exception, TNotice> errorNoticeSelector,
-            TNotice endNotice,
-            int maxConcurrency,
-            CancellationTokenSource cancellationTokenSource)
+        enum Notice { End, Result, Error }
+
+        static async Task StartAsync<T, TResult>(
+            this IEnumerator<T> enumerator,
+            Func<T, Task<TResult>> starter,
+            Action<T, Task<TResult>> onCompletion,
```

Comment:
I can understand why you'd say that and you'll notice that I got side-tracked in d973670617ca0393b576204457945ef139541283 by the observer pattern here. Later in da262bc8c1e58295a741f33f444862f23b50f1e8, I reverted (as allowing harmless race conditions meant I couldn't guarantee the pattern's semantics). I find `onNext` will mislead in the same direction once again. Tasks _complete_ so completion here is in that sense rather than completion of the loop. I felt that the argument containing a task would be sufficient to say what's completing in case there's any confusion but perhaps that's not how you read it. I would prefer `onTaskCompletion` than `onNext` if you still feel strongly about it.

Reply:
OK, I see why you don't want `onNext`. `onTaskCompletion` would be a better name. The main issue I think is that `onCompletion` is too easy to confuse with completing *the entire* task instead of completing *each iteration*. But your proposed name works for me.